src/main/java/uk/ac/ebi/ampt2d/metadata/persistence/services/StudyServiceImpl.java

```diff
@@ -17,24 +17,70 @@
  */
 package uk.ac.ebi.ampt2d.metadata.persistence.services;
 
+import com.fasterxml.jackson.databind.ObjectMapper;
 import com.querydsl.core.types.Predicate;
 import org.springframework.beans.factory.annotation.Autowired;
+import uk.ac.ebi.ampt2d.metadata.persistence.entities.AccessionVersionEntityId;
 import uk.ac.ebi.ampt2d.metadata.persistence.entities.QStudy;
 import uk.ac.ebi.ampt2d.metadata.persistence.entities.Study;
 import uk.ac.ebi.ampt2d.metadata.persistence.repositories.StudyRepository;
 
+import java.time.LocalDate;
+import java.util.Comparator;
 import java.util.List;
 
 public class StudyServiceImpl implements StudyService {
 
     @Autowired
     private StudyRepository studyRepository;
 
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Override
+    public Study findOneStudyByPredicate(Predicate predicate) {
+        return studyRepository.findOne(predicate);
+    }
+
+    @Override
+    public Study findOneStudyByAccession(AccessionVersionEntityId id) {
+        QStudy study = QStudy.study;
+        Predicate predicate = study.id.accession.equalsIgnoreCase(id.getAccession())
+                .and(study.id.version.eq(id.getVersion()));
+
+        return findOneStudyByPredicate(predicate);
+    }
+
     @Override
     public List<Study> findStudiesByPredicate(Predicate predicate) {
         return (List<Study>) studyRepository.findAll(predicate);
     }
 
+    @Override
+    public List<Study> findStudiesByAccession(String accession) {
+        QStudy study = QStudy.study;
+        Predicate predicate = study.id.accession.equalsIgnoreCase(accession);
+
+        List<Study> studies = findStudiesByPredicate(predicate);
+
+        studies.sort((new Comparator<Study>() {
+            @Override
+            public int compare(Study o1, Study o2) {
+                return o1.getId().getVersion() - o2.getId().getVersion();
+            }
+        }).reversed());
+
+        return studies.subList(0, studies.size() > 0 ? 1 : 0);
```

Comment:
According to the API documentation this method returns only the last version of an accession so:
* It should be renamed to make it clear
* The return type should be changed to `Study`
* The last call would then be

```java
return studies.isEmpty() ? null : studies.get(0);
```

Reply:
Fixed in 4b80438